src/lib/validateGameFiles.ts

```diff
@@ -0,0 +1,241 @@
+import { ESLint, Linter } from "eslint"
+import { parse, valid } from "node-html-parser"
+import semver from "semver"
+
+import { extractMultiplayerMetadata } from "./extractMultiplayerMetadata.js"
+import { FileInfo } from "./getGameFiles.js"
+
+import LintMessage = Linter.LintMessage
+
+export const validationOptions = {
+  sdkUrlStart: "https://cdn.jsdelivr.net/npm/rune-games-sdk",
+  sdkVersionRegex: /rune-games-sdk@(\d\.\d(\.\d)?)/,
+  minSdkVersion: "2.4.0",
+  maxFiles: 1000,
+  maxSizeMb: 25,
+}
+
+const eslint = new ESLint({
+  useEslintrc: false,
+  allowInlineConfig: false,
+  baseConfig: {
+    root: true,
+    extends: ["plugin:rune/logic"],
+  },
+})
+
+export interface ValidationError {
+  message: string
+  lintErrors?: LintMessage[]
+}
+
+export interface ValidationResult {
+  valid: boolean
+  errors: ValidationError[]
+  multiplayer:
+    | {
+        minPlayers?: number
+        maxPlayers?: number
+        handlesPlayerJoined?: boolean
+        handlesPlayerLeft?: boolean
+      }
+    | undefined
+}
+
+export async function validateGameFiles(
+  files: FileInfo[]
+): Promise<ValidationResult> {
+  const { sdkUrlStart, sdkVersionRegex, minSdkVersion, maxFiles, maxSizeMb } =
+    validationOptions
+
+  const errors: ValidationError[] = []
+
+  if (files.length > maxFiles) {
+    errors.push({ message: `Too many files (>${maxFiles})` })
+  }
+
+  const totalSize = files.reduce((acc, file) => acc + file.size, 0)
+
+  if (totalSize > maxSizeMb * 1e6) {
+    errors.push({ message: `Game size must be less than ${maxSizeMb}MB` })
+  }
+
+  const indexHtml = files.find(
+    (file) => file.path === "index.html" || file.path.endsWith("/index.html")
+  )
+  const logicJs = files.find(
+    (file) => file.path === "logic.js" || file.path.endsWith("/logic.js")
+  )
+
+  let multiplayer: ValidationResult["multiplayer"]
+
+  if (indexHtml) {
```

Comment:
Can we rewrite this into a validator with rules which would iterate through them instead of having n levels deep if elses? I can't review this code.